sematic-ai/sematic
# Expose the Cloned Run ID in the UI

This is the first in a series of PRs that will implement #338.

For cloned Runs, it persists and exposes the ID of the original Run it was cloned from, and also mentions the cloning as the reason the logs are absent. This will be used for both the "rerun-from-here" and caching features.

UI rendering of the original run:
<img width="1395" alt="image" src="https://user-images.githubusercontent.com/1894533/207347240-f5612d35-5edc-47ee-b3c0-ae83bc302b28.png">

UI rendering of the cached run:
![image](https://user-images.githubusercontent.com/1894533/207347975-76e09a82-6629-436d-9ad3-79937e390ec0.png)

## Testing

### Unit Tests

Updated the unit tests, and added a new test case for the log reader.

### Manual Test Cases

Verified on a "rerun-from-here" pipeline that the runs that should have been invalidated don't show original run id entries in the UI, and that the cloned ones do show original run id entries.

Verified that pipelines that have been reran from also-rerun pipelines show the original ids, and not the intermediate ones:
<img width="378" alt="image" src="https://user-images.githubusercontent.com/1894533/207348342-da22367a-72f0-403a-9b03-7620d9913b0b.png">
<img width="563" alt="image" src="https://user-images.githubusercontent.com/1894533/207348428-d77863af-0fa2-47d8-b0f0-59843c1e8d15.png">
<img width="563" alt="image" src="https://user-images.githubusercontent.com/1894533/207348523-c6f3aecb-5669-43f9-a53a-f6fc5012995d.png">
